matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -119,12 +119,15 @@ public MapMatching(GraphHopper graphHopper, PMap hints) {
         RoutingAlgorithmFactory routingAlgorithmFactory = graphHopper.getAlgorithmFactory(profile.getName(), disableCH, disableLM);
         if (routingAlgorithmFactory instanceof CHRoutingAlgorithmFactory) {
             ch = true;
-            routingGraph = graphHopper.getGraphHopperStorage().getCHGraph(((CHRoutingAlgorithmFactory) routingAlgorithmFactory).getCHProfile());
+            CHProfile chProfile = ((CHRoutingAlgorithmFactory) routingAlgorithmFactory).getCHProfile();
+            routingGraph = graphHopper.getGraphHopperStorage().getCHGraph(chProfile);
+            weighting = chProfile.getWeighting();
```

Comment:
Its important to use the same weighting that was used for the CH preparation. This is what we do for routing as well currently.

Reply:
Ok the problem here is that this way we are ignoring the hints (most importantly headingPenalty) -> use createWeighting again